async/actorpromise.md

```diff
@@ -0,0 +1,127 @@
+    
+  be receive(s: A) =>
+    _coll.push(consume s)    
+    if _coll.size() == _max then      
+      let output = _coll = recover Array[A] end
+      _target.receivecollection(consume output)
+    end
+```
+Now that we have an intermediary, we can create multiple promises to send to multiple bank accounts:
+```pony
+let accounts = ["0001"; "0002"; "0003"; "0004"]
+let collector = Collector[AccountSummary](accounts.size(), this)
+    
+for account in accounts.values() do
+  let aggregate = AccountAggregate(account, 6000)
+  let p = Promise[AccountSummary]
+  p.next[None](recover collector~receive() end)
+  aggregate.summarize(p)
+end 
+```
+Our bank account aggregate can be modified to include an account summary with the **summarize** method:
+
+```pony
+be summarize(p: Promise[AccountSummary]) =>    
+  p(recover AccountSummary(_balance, _account) end)
+```
+
+Finally, we add the behavior that the collector is expecting to the calling actor in order to respond to the list of account summaries:
+
+```pony
+be receivecollection(coll: Array[AccountSummary] val) =>
+  _env.out.print("received account summaries:")
+  for summary in coll.values() do
+    _env.out.print("Account " + summary.accountnumber() + ": $" + summary.currentbalance().string())
+  end
+```
+## Discussion
+Actor systems have been around for quite some time now, but most developers don't default to modeling their problems as actor patterns. Most of us want to solve this problem with synchronous code that looks like this:
+
+```pony
+for acct in _accounts.values() do
+    _summaries.push(acct.summarize())
+end
+```
+Truthfully this is probably a good way to solve a lot of problems. In the trivial example discussed here, we're not actually gaining all that much from the use of actors, so why incur all that extra complexity? Consider these questions:
+
+What if the `balance` wasn't a cached field; what if we had to run through the list of events and calculate it on the fly every time a consumer asked for the balance? What if part of the account summary included a test for reconciliation and pending transactions that might incur a network call to an out-of-process cache or a microservice?
+
+Now there's a _cost_ associated with asking a bank account to provide a summary. Just running through the summarization method in a for loop can now cause a consumer to wait an indeterminate amount of time. By sending out a flood of promises, we can let each of the actors fulfill the promise on their own time and we'll get the results back far sooner than if we'd done the requests synchronously. This also gives us an added degree of reliability - by sending out these promises, we can also set a timeout in the collector so that we can build in things like a "circuit breaker" and return data indicating that we couldn't summarize all of the accounts.
+
+You can find a working example of the code for this pattern in [this repository](https://github.com/autodidaddict/bank-pony)
```

Comment:
I'd like to keep these patterns self contained. When we make breaking changes, we won't be able to update that repo w/ your assistance. Can you find a way to make this a more self contained example?